src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -626,7 +626,7 @@ bool is_numeric_property_value_superset(
           json::Value{prop_name.data(), rapidjson::SizeType(prop_name.size())});
 
         if (it == v.MemberEnd()) {
-            return default_value;
+            return std::nullopt;
```

Comment:
Otherwise I can't tell further down whether the value is unset or set to the default value. And I need to know that because there's a specific error code for "this-property-was-added".

Now instead of falling back to the default value here, I handle comparing the value against the default further down.